### PR TITLE
Add new 'invasions' DTS and filters and fetch 'invasions' automatically

### PR DESCRIPTION
**Pull request by Pontiky**
## Description

General changes
- Fetch automatically invasions data (from RocketMAD source) without making a new PR. This change follow PR #802 and #813.
- Add a lot of DTS and filters (see below)
- Update 'invasions' doc

Renamed DTS
- `type_id` -> `grunt_id` : Grunt ID.

Added DTS
- `grunt_id_3` : Grunt ID, padded to 3 digits. (still existing but not referenced in the doc)
- `grunt_name` : Name of the grunt (ex: Sierra, Giovanni, Male...)
- `type_name` : Name of the monster's type used by the grunt.
- `type_emoji` : Emoji for monster's type, or empty string if unknown.
- `reward_ids` : List of possible rewarded monsters ID.
- `reward_names` : List of possible rewarded monsters.
- `battle1_ids` : List of possible monsters ID for the 1st battle.
- `battle1_names` : List of possible monsters for the 1st battle.
- `battle2_ids` : List of possible monsters ID for the 2nd battle.
- `battle2_names` : List of possible monsters for the 2nd battle.
- `battle3_ids` : List of possible monsters ID for the 3rd battle.
- `battle3_names` : List of possible monsters for the 3rd battle.

Renamed filters
- `grunt_types` -> `grunt_ids` : Array of allowed grunts, by id.
- `grunt_types_exclude` -> `grunts_exclude` : Array of excluded grunts, by id.

Added filters
- `types` : Array of allowed monster types, by name.
- `monsters` : Array of allowed monsters possibly rewarded, by  or name.
- `monsters_exclude` : Array of excluded monsters possibly rewarded, by id or name.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (would cause existing functionality to change)

## Motivation and Context

The changes will break the current Invasions configurations as it renames some existing DTS and filters to stop the confusion between grunt types (id of the grunt) and monster type used by the grunt. The doc was wrong as it mixed up these 2 stats.

The new DTS and filters will allow us to correctly filters and display invasion informations.

## How Has This Been Tested?

New unit tests and live configuration tests:
image

## Wiki Update

- [x] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

### CODE REVIEW

**Review by billyjbryant — CHANGES_REQUESTED** [review]
Please rebase and resolve conflicts

File: start_pokealarm.py [hunk 1]
```diff
@@ -285,12 +298,53 @@ def download_data(sigdiff=None):
...
+        os.replace("data/tmp_invasions.json", "data/invasions.json")
 
 # Configure and run PokeAlarm
+
+
 def start_server():
     # Parse Settings
     parse_settings(os.path.abspath(os.path.dirname(__file__)))
 
+    # import PokeAlarm.Utilities.GruntUtils as gu
```
Comment (LegitDongo) [inline start_pokealarm.py]: Forgot to remove some test code

**Review by LegitDongo — CHANGES_REQUESTED** [review]
Small changes, haven't tested

This will probably need a rebase after #807 gets merged, which I intend to merge first given public testing goes ok

**Comment by Pontiky** [comment]
About the last commit I made, the invasion tests work as expected but given the Pokemon pool of grunts often changes without using a new grunt ID, these tests used to check if a grunt is using a specific Pokemon or type can change anytime, breaking the tests by giving false negative results and causing the failure of the CI for the next pull requests.

If some changes are made to invasions, please uncomment these tests and update the ID depending on the current pool of each grunt. https://raw.githubusercontent.com/cecpk/RocketMAD/master/static/data/invasions.json

It doesn't change anything else. I've still proven that these tests work.

**Review by LegitDongo — APPROVED** [review]
Looks pretty good

My only complaint is that the naming seems inconsistent, but I'm going ahead with the merge

Side note: production tests confirmed

**Pull request merged**